persist/sqldb/workflow_archive.go

```diff
@@ -158,7 +156,7 @@ func (r *workflowArchive) ListWorkflows(namespace string, name string, namePrefi
 	}
 
 	err = r.session.
-		Select("name", "namespace", "uid", "phase", "startedat", "finishedat").
```

Comment:
hmm this might be part of the performance regression per #12025, or rather #12030 more specifically. In particular, if this pulls the `status`, it could be huge -- even 20 archived workflows could total more than a 20MB response from the DB to the Server

Reply:
This performance regression seems to have been confirmed by a user [on Slack](https://cloud-native.slack.com/archives/C01QW9QSSSK/p1706628618551249)